api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -439,6 +441,28 @@ func validateLoadBalancerProfile(loadBalancerProfile *LoadBalancerProfile, fldPa
 	return allErrs
 }
 
+func validateAMCPVirtualNetwork(amcpSpec AzureManagedControlPlaneSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// amcpSpec.VirtualNetwork.ResourceGroup is not set, it means the virtual network is in the same resource group as the control plane.
```

Comment:
I'm a little bit confused about this comment. It looks like the if statement is checking if the vnet is in a different resource group, and it doesn't check if it's not set. Should we return early if it's not set?

Reply:
This is a ghost comment, should have removed it. Sorry about the confusion. 😅 
What I really meant to convey was that if `amcpSpec.VirtualNetwork.ResourceGroup` is not set, if gets defaulted to `amcpSpec.ResourceGroupName` in the defaulting webhook. 
But that comment would have been and is kinda redundant.